src/kernl/implementations/cuda_graph.py

```diff
@@ -12,31 +12,77 @@
 #  See the License for the specific language governing permissions and
```

Comment:
I don't know why this file is in this folder, it should be in optimizer, maybe mistake in previous commits

Reply:
done